test/evp_kdf_test.c

```diff
@@ -253,7 +253,7 @@ static int do_kdf_hkdf_gettables(int expand_only, int has_digest)
                                                  "secret", 6, "salt",
                                                  expand_only ? NULL : "label"))
         || !TEST_ptr(kctx = get_kdfbyname(OSSL_KDF_NAME_HKDF))
-        || !TEST_true(EVP_KDF_CTX_set_params(kctx, params)))
+        || TEST_int_le(EVP_KDF_CTX_set_params(kctx, params), 0))
```

Comment:
I'd prefer the `|| !TEST_xxx` form here since we use it fairly universally in strings of OR operations.

Reply:
Yeah, this is a wrong form as it will report error for expected failure of the call.

Reply:
@PeiweiHu please fix these checks